### PR TITLE
Added the perceptron learner algorithm

### PR DESCRIPTION
**Pull request by AbdulazizAlaa**
Added the code for the perceptron learner. It contains several functions with their comments. 

the most important are: gradientDescient an implementation of the gradient descent that loops for a specific number of iterations and every time computes the cost and gradient, then it updates the weights using the gradient. cost a function that computes the cost and gradient using mathematical formal. generateTargetList converts the target list into a list of 1 and -1 to denote the existence of some class.

i hope everything is clear.

### CODE REVIEW

**Comment by Schrodinger1926** [comment]
Hey @AbdulazizAlaa , Perceptron has already been implemented. You might not have pulled latest commits from upstream master. I see you wrote an independent patch here, we need to move according to aima as well keep neural net learner and perceptron learner in sync by using same underlying backpropagation algorithm.

**Comment by AbdulazizAlaa** [comment]
Hi @Schrodinger1926, yeah i see it now.  Thank you. Can you guide me to what issue maybe of benefit to contribute to. Thank you again.

**Comment by Chipe1** [comment]
@AbdulazizAlaa checkout the readme.MD for a list of unimplemented algorithms to implement. You can also write Ipython jupyter notebooks :)

**Comment by Schrodinger1926** [comment]
@AbdulazizAlaa that's right
